bundles/org.openhab.binding.plugwiseha/src/main/java/org/openhab/binding/plugwiseha/internal/handler/PlugwiseHABridgeHandler.java

```diff
@@ -0,0 +1,282 @@
+        } else {
+            logger.warn("Invalid config for the Plugwise Home Automation bridge handler with config = {}", this.config);
+        }
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(PlugwiseHADiscoveryService.class);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        this.logger.warn(
+                "Ignoring command = {} for channel = {} - this channel for the Plugwise Home Automation binding is read-only!",
+                command, channelUID);
+    }
+
+    @Override
+    public void dispose() {
+        cancelRefreshJob();
+        if (this.controller != null) {
+            this.controller.stop();
+            this.controller = null;
+        }
+    }
+
+    public static boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_BRIDGE_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    // Getters & setters
+
+    public @Nullable PlugwiseHAController getController() {
+        return this.controller;
+    }
+
+    // Protected and private methods
+
+    /**
+     * Checks the configuration for validity, result is reflected in the status of
+     * the Thing
+     */
+    private boolean checkConfig() {
+        if (this.config == null || !this.config.isValid()) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, STATUS_DESCRIPTION_CONFIGURATION_ERROR);
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    private void scheduleRefreshJob() {
+        synchronized (this) {
+            if (this.refreshJob == null) {
+                logger.debug("Scheduling refresh job every {}s", config.getRefresh());
+                this.refreshJob = scheduler.scheduleWithFixedDelay(this::run, 0, config.getRefresh(), TimeUnit.SECONDS);
+            }
+        }
+    }
+
+    private void run() {
+        try {
+            logger.trace("Executing refresh job");
+            refresh();
+            updateStatus(ONLINE);
+        } catch (PlugwiseHAInvalidHostException e) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, STATUS_DESCRIPTION_INVALID_HOSTNAME);
+        } catch (PlugwiseHAUnauthorizedException | PlugwiseHANotAuthorizedException e) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, STATUS_DESCRIPTION_INVALID_CREDENTIALS);
+        } catch (PlugwiseHACommunicationException e) {
+            updateStatus(OFFLINE, COMMUNICATION_ERROR, STATUS_DESCRIPTION_COMMUNICATION_ERROR);
+        } catch (PlugwiseHATimeoutException e) {
+            updateStatus(OFFLINE, COMMUNICATION_ERROR, STATUS_DESCRIPTION_TIMEOUT);
+        } catch (PlugwiseHAException e) {
+            logger.debug("Unhandled exception while refreshing the Plugwise Home Automation Controller {} - {}",
+                    getThing().getUID(), e.getMessage());
+            updateStatus(OFFLINE, COMMUNICATION_ERROR, e.getMessage());
+        }
```

Comment:
You might want to catch `RuntimeException` to catch unexpected errors. Sometimes due to a communication problem something can go wrong. And if you don't catch `RuntimeException` it will stop the refresh process and the binding has to be restarted by the user to get things going.

Reply:
Wat would the right level be? warn?

Reply:
In such cases I set the thing to offline with the exception message as description and log the exception to debug.